silq/instrument_interfaces/keysight/E8267D_interface.py

```diff
@@ -258,6 +276,23 @@ def get_additional_pulses(self, connections) -> List[Pulse]:
         for pulse in self.pulse_sequence:
             additional_pulses += pulse.implementation.get_additional_pulses(interface=self)
 
+        # Ensure marker pulses are not overlapping
+        marker_pulses = [p for p in additional_pulses if isinstance(p, MarkerPulse)]
+        if marker_pulses:
+            marker_pulses = sorted(marker_pulses, key=lambda p: p.t_start)
+            current_pulse = marker_pulses[0]
+            merged_marker_pulses = [current_pulse]
+            for pulse in marker_pulses[1:]:
+                if pulse.t_start <= current_pulse.t_stop:
+                    # Marker pulse overlaps with previous pulse
+                    current_pulse.t_stop = max(pulse.t_stop, current_pulse.t_stop)
+                else:
+                    # Pulse starts after previous pulse
+                    merged_marker_pulses.append(pulse)
+                    current_pulse = pulse
+            nonmarker_pulses = [p for p in additional_pulses if not isinstance(p, MarkerPulse)]
+            additional_pulses = [*merged_marker_pulses, *nonmarker_pulses]
+
```

Comment:
I am a bit afraid that if envelope_padding has "large" value, and general (not marker) pulses have "small" inter-delays, there will be lots of overlaps there since you sort only marker pulses...

Reply:
I don't follow your reasoning, we should discuss

Reply:
Can we resolve this conversation?

Reply:
yes